diff --git a/includes/menu.inc b/includes/menu.inc
index a93ddad..fe72390 100644
--- a/includes/menu.inc
+++ b/includes/menu.inc
@@ -1509,7 +1509,11 @@ function _menu_tree_check_access(&$tree) {
   $new_tree = array();
   foreach ($tree as $key => $v) {
     $item = &$tree[$key]['link'];
-    _menu_link_translate($item);
+    // Prevent translating menu links twice.
+    if (!isset($item['menu_link_translated'])) {
+      _menu_link_translate($item);
+      $item['menu_link_translated'] = TRUE;
+    }
     if ($item['access'] || ($item['in_active_trail'] && strpos($item['href'], '%') !== FALSE)) {
       if ($tree[$key]['below']) {
         _menu_tree_check_access($tree[$key]['below']);
